Add rfc 0016 automated verification of node compliance (#53)
internet-sicherheit/eco-blockchain-governance: technical/0016-automated-verification-of-node-compliance/README.md

# 0016: Automated verification of node compliance
- Authors: [David Maas](david_maas@hotmail.de), [Mirko Mollik](mollik@trustcerts.de) 
- Status: PROPOSED
- Since: 2020-09-09 
- Status Note: Initial Proposal  
- Start Date: 2020-09-09 
- Tags: 

## Summary
Describes the automatic verification of nodes with a specific compliance.

## Motivation

> To operate a safe environment at all times, it is important that the nodes meet certain requirements. Since a manual process would be too time-consuming, an automation of Verified Nodes is very important. Thereby any compliance can be defined.

Reviewer: ```suggestion
To operate a safe environment at all times, it is important that the nodes meet certain requirements. 
Since a manual process would be too time-consuming, automation of Verified Nodes is very important. 
Thereby any compliance should be defined.
```

## Specification

> To enable verification, the following points would have to be present.

Reviewer: ```suggestion
To enable verification, the following points would have to be present:
```

* Compliance guidelines

> * Observer

Reviewer: What does observer mean?

Author: In that context, it would be watchdog routines. So that the compliance with the regulations is continuous.

* Verification routines
* Time interval of the repetition

## Examples & Best Practices

> [Hyperledger Fabric](https://hyperledger-fabric.readthedocs.io/en/release-2.2/policies/policies.html) provides basic ways to apply compliance to a Blockchain.

Reviewer: As I see it, Fabric only defines compliance on the protocol level.

Author: The architecture is going in that direction. But not everything has been defined yet in terms of compliance.

>Policies are one of the things that make Hyperledger Fabric different from other blockchains like Ethereum or Bitcoin. In those systems, transactions can be generated and validated by any node in the network. The policies that govern the network are fixed at any point in time and can only be changed using the same process that governs the code. Because Fabric is a permissioned blockchain whose users are recognized by the underlying infrastructure, those users have the ability to decide on the governance of the network before it is launched, and change the governance of a running network.

## Drawbacks

> If security gaps could be exploited, automation would of course also be very dangerous.

Reviewer: This drawback seems arbitrary.

Author: When processes are automated, the exploitation of security vulnerabilities with the subsequent execution of non-legitimate measures is of course also problematic. That is what is meant by this. But that is also very general.
